### DISCUSSION TITLE
Requiring commit message style

### DISCUSSION THREAD

**Discussion by coolapso** [pivpn/pivpn#1578] [Feature requests]
in sequence to this one: https://github.com/pivpn/pivpn/discussions/1573 

I think its worth also requiring commit message styles. 

I suggest this one: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#-commit-message-format

Which in turn allows the usage of this: https://semantic-release.gitbook.io/semantic-release/

This has been in use already for the Docs repository, and can see it in action here: https://github.com/pivpn/docs/releases 

**Bennefits:** 
* PiVPN starts having versioning, and its done automatically.  not that it matters much for PiVPN but its nice to have. 
* The Change log gets generated automatically and we do not have to edit it every-time before merging to master. 
* Every commit message is done the same way, easy to identify and read what changed. 

Tagging currently active contributors: @orazioedoardo @giulio-coa @MichaIng @TinCanTech

**Comment by ghost** [comment]
Ok
I will start from the next commit

**Reply by coolapso** [reply]
Once all the new changes are done, i will take care of the CI, and most likely move it to GH Actions as well.

**Comment by ghost** [comment]
#1572

**Reply by coolapso** [reply]
yup looks about right :) ty!
